woocommerce/woocommerce-google-analytics-integration
# E2E tests for the All Products block

### Changes proposed in this Pull Request:

This PR adds two additional E2E tests for the "All Products" block. Initially this block was used for the default Shop page, but that depends on which WooCommerce version is used. With the current version of WooCommerce the Shop page defaults to using the "Products (Beta)" block. So this PR ensures we explicitly test the block type regardless of which block is used for the shop page.

Like the other block page tests this currently tests two things:
1. Adds a product to the cart from the All Products block page
2. Checks that the event `view_item_list` is sent from the All Products block page

> [!Note]
The All Products block does not set a data attribute for each product so we can't select a specific add to cart button. However since the results are sorted by latest first we can pick the first add to cart button from the page.

This resolves one of the testing points in #248

### Detailed test instructions:

1. Install all npm dependencies `nvm use && npm ci`
2. Build assets either through `npm start` or `npm run build`
3. Start the wp-env site npm run wp-env:up
4. Install browser (only needed once) `npx playwright install chromium`
5. Run tests npm run test:e2e
6. Confirm tests are successful (in particular the added All Products block tests)

### Changelog entry
* Dev - E2E tests for the All Products block.
